src/ispc/kernels/lanczos3.ispc

```diff
@@ -76,7 +76,12 @@ static inline uint8<4> resample_internal(uniform Image src_image, float<2> uv, u
 
             float<4> texel;
 
-            if (num_channels == 3) {
+            if (num_channels == 1) {
+                texel.x = byte_to_float(src_image.data[addr + 0]) * inv_255;
+            } else if (num_channels == 2) {
+                texel.x = byte_to_float(src_image.data[addr + 0]) * inv_255;
+                texel.y = byte_to_float(src_image.data[addr + 1]) * inv_255;
+            } else if (num_channels == 3) {
```

Comment:
That's going to be a fun conflict with #15

Reply:
Shouldn't be too hard. Add `clean_and_write_2_channels` and `clean_and_write_1_channel`..at least from a quick glance it looks like that's the way to go?
Anyway I'm just noticing that also `intel_tex_2` pretty much relies on `RgbaSurface`s... so maybe this change is an overkill?

Reply:
> Shouldn't be too hard. Add `clean_and_write_2_channels` and `clean_and_write_1_channel`..at least from a quick glance it looks like that's the way to go?

It's a bit much to have those functions for every component count, I wish we could do something simpler and have the compiler "optimize it away" but there were some specific details...

> Anyway I'm just noticing that also `intel_tex_2` pretty much relies on `RgbaSurface`s, so maybe this change may end up not needed after all :sweat_smile: 

As in shouldn't we update `intel_tex_2` then?